### PR TITLE
Update idl conversion script to new webidl2js

### PR DESCRIPTION
**Pull request by Sebmaster**

### CODE REVIEW

File: scripts/webidl/convert.js [hunk 1]
```diff
@@ -1,56 +1,23 @@
...
-  });
+function addDir(dir) {
+  const resolved = path.resolve(__dirname, dir);
+  transformer.addSource(resolved, resolved);
 }
 
-function onlyIDL(filePath) {
-  return path.extname(filePath) === ".idl";
-}
+addDir(path.resolve(__dirname, "../../lib/jsdom/living/traversal"));
```
Comment (domenic) [inline scripts/webidl/convert.js]: No need to double-resolve

File: scripts/webidl/convert.js [hunk 2]
```diff
@@ -1,56 +1,23 @@
...
+const Webidl2js = require("webidl2js");
 
-  return Q.nfcall(fs.stat, inputPath)
-  .then(inputStat => {
-    isDir = inputStat.isDirectory();
-    if (isDir) {
-      return readdirRecursive(inputPath, onlyIDL).map(relativePath => path.resolve(inputPath, relativePath));
-    }
+const transformer = new Webidl2js();
+transformer.options.suppressErrors = true;
```
Comment (domenic) [inline scripts/webidl/convert.js]: This seems like a bad API; the options should be passed in to the constructor, I think.

**Comment by Sebmaster** [comment]
Fixed both of these up. Ready for webidl2js 5.0? :smile:

File: scripts/webidl/convert.js [hunk 3]
```diff
@@ -1,56 +1,28 @@
...
-  .done();
+addDir("../../lib/jsdom/living/traversal");
+addDir("../../lib/jsdom/living/events");
+addDir("../../lib/jsdom/living/attributes");
+addDir("../../lib/jsdom/living/window");
+addDir("../../lib/jsdom/living/nodes");
+addDir("../../lib/jsdom/living/navigator");
+
+transformer.generate(path.resolve(__dirname, "../../lib/jsdom/living/generated/"))
+  .catch ((err => {
```
Comment (domenic) [inline scripts/webidl/convert.js]: Extra space after `.catch `, extra parens around arrow function.

**Comment by domenic** [comment]
LGTM with formatting nit. Also needs a rebase I guess. Publish 5.0.0/merge at will!

**Comment by domenic** [comment]
@Sebmaster ping on publishing webidl2js 5.0.0?

**Comment by snuggs** [comment]
How are wae looking on this folks @domenic @Sebmaster? Also would you mind pointing me to a good reference for webidl2js and what problem it solves? Never heard of it.

**Comment by snuggs** [comment]
NVM Got it! I read tons of DOM specs so now have an understanding of what i've been reading. https://en.wikipedia.org/wiki/Web_IDL

**Comment by snuggs** [comment]
How do ya use that thing? Documentation is sparse and took a bit to figure out the jsdom group had it. There's also no tests. 
¯\_(ツ)_/¯ https://github.com/jsdom/webidl2js 

I see it's used in jsdom core and have some upcoming work i'd like to test out on a branch with Web Audio. Have a WebRTC app and would like to make additions the "jsdom" way. Any help would be appreciated! /cc @domenic

**Comment by Sebmaster** [comment]
@domenic Should be good to go now, I think.

@snuggs The code in this PR is the the only doc (for now). If it's for contributing to jsdom, copy the IDL snippet from the specs, put it in the appropriate dir (or create a new one and add it to the conversion script) and create impl classes, done.

**Pull request merged**